### ISSUE TITLE
Split up in multiple packages

### ISSUE THREAD

**Issue by icidasset** [oddsdk/ts-odd#571]
## Problem

Instead of using peer dependencies, code should be split up in multiple packages. There are a lot of dependencies that are component specific. For example, wether or not you use IPFS/libp2p.

## Solution

Split up in multiple packages.

I'm not entirely sure what the best way is to split these up. Maybe we start with splitting up the **Core**: Everything that's in the SDK now, except the Fission stuff. We do want basic components in there because they will be reused a lot in combination with the non-core components. This includes things like storing session data in indexedDB. Local-only mode components for everything are included here too, because they are so basic.

From there we could have packages for Fission and Web3Storage. These involve the account system and block storage. We also have different kinds of packages, those that involve the identifier component, passkeys and walletauth. I'm not sure if you can call these plugins or not?

TLDR:
- Core
- Fission, Web3Storage, other account/storage layers.
- Passkeys, walletauth, other identifier methods.
- CLI

Could definitely be split up more. But that seems like a good place to start?
